### PR TITLE
go: Add structuredheader.ParseListOfLists

### PR DESCRIPTION
**Pull request by irori**
This will be used to parse Variant / Variant-Key headers.

### CODE REVIEW

File: go/signedexchange/structuredheader/parser.go [hunk 1]
```diff
@@ -96,6 +116,31 @@ func (p *parser) parseKey() (Key, error) {
 	}
 	return Key(p.getString(i)), nil
 }
+// https://tools.ietf.org/html/draft-ietf-httpbis-header-structure-09#section-4.2.4
+func (p *parser) parseListOfLists() (ListOfLists, error) {
+	var top_list ListOfLists
```
Comment (nyaxt) [inline go/signedexchange/structuredheader/parser.go]: Style nit: https://golang.org/doc/effective_go.html#mixed-caps

Note to self: I was going to comment `innerList := []Item{}`, but found https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

File: go/signedexchange/structuredheader/parser.go [hunk 2]
```diff
@@ -138,7 +183,7 @@ func (p *parser) parseParameterisedIdentifier() (ParameterisedIdentifier, error)
 		if _, ok := parameters[param_name]; ok {
 			return ParameterisedIdentifier{}, fmt.Errorf("structuredheader: duplicated parameter '%s'", param_name)
 		}
-		var param_value interface{}
+		var param_value Item
```
Comment (nyaxt) [inline go/signedexchange/structuredheader/parser.go]: subCamelCase (for other vars too. sorry, should have caught in earlier reviews)

File: go/signedexchange/structuredheader/parser.go (hunk 1 above)
Comment (irori) [reply go/signedexchange/structuredheader/parser.go]: Done

File: go/signedexchange/structuredheader/parser.go (hunk 2 above)
Comment (irori) [reply go/signedexchange/structuredheader/parser.go]: Oops, done.

**Pull request merged**